src/SLIP10Node.ts

```diff
@@ -276,23 +280,27 @@ export class SLIP10Node implements SLIP10NodeInterface {
   }
 
   public get chainCode() {
-    return this.chainCodeBuffer.toString('hex');
+    return bytesToHex(this.chainCodeBuffer);
   }
 
   public get privateKey(): string | undefined {
-    return this.privateKeyBuffer?.toString('hex');
+    if (this.privateKeyBuffer) {
+      return bytesToHex(this.privateKeyBuffer);
+    }
+
+    return undefined;
   }
 
   public get publicKey(): string {
-    return this.publicKeyBuffer.toString('hex');
+    return bytesToHex(this.publicKeyBuffer);
```

Comment:
All of these will be 0x prefixed now, if that is intended we should make sure to let Snap developers know when we ship this.

Reply:
That's intentional, and yeah, we need to communicate that.

I think in general we should standardise hexadecimal values everywhere to start with `0x`. The problem is otherwise that we have to guess what the user meant. Values like `0x1234` are clearly hexadecimal, but `1234` could be interpreted as a number as well. The same goes for UTF-8-like values.

Reply:
I second this. If it ain't `0x`-prefixed it better be a local variable with a descriptive name.